pkg/workloadmeta/collectors/internal/containerd/image.go

```diff
@@ -0,0 +1,344 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build containerd
+// +build containerd
+
+package containerd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
+	log "github.com/cihub/seelog"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/api/events"
+	"github.com/containerd/containerd/content"
+	containerdevents "github.com/containerd/containerd/events"
+	"github.com/containerd/containerd/images"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/gogo/protobuf/proto"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const imageTopicPrefix = "/images"
+
+func isImageTopic(topic string) bool {
+	return strings.HasPrefix(topic, imageTopicPrefix)
+}
+
+func (c *collector) handleImageEvent(ctx context.Context, containerdEvent *containerdevents.Envelope) error {
+	switch containerdEvent.Topic {
+	case imageCreationTopic:
+		event := &events.ImageCreate{}
+		if err := proto.Unmarshal(containerdEvent.Event.Value, event); err != nil {
+			return fmt.Errorf("error unmarshaling containerd event: %s", err)
```

Comment:
Just a small tip to nest errors.

```suggestion
			return fmt.Errorf("error unmarshaling containerd event: %w", err)
```